config/routes.rb

```diff
@@ -4,16 +4,17 @@
   end
 
   devise_for :users, skip: [:registrations, :confirmations, :sessions, :passwords], skip_helpers: true
-  
+
   namespace :api do
     namespace :v1 do
       use_doorkeeper do
         controllers tokens: 'tokens'
         skip_controllers :authorizations, :applications, :authorized_applications, :token_info, :tokens
       end
+
       resources :keywords, only: :index
       resources :private_items, only: :index
-      resources :tokens, only: [:create]
+      post '/signup', to: 'registrations#create'
```

Comment:
```suggestion
      resources :registrations, only [:create]
```

As said earlier, unless specifications or client constrains the project to use a custom route, when working with Rails we have a lot to gain if we stick to the default resources routes. Because Rails mindset is `convention over configuration`, if you follow the conventions (routes must follow the model name, controller/actions, only `index/new/edit/create/update/destroy` actions, etc...), then most of the configurations will be done implicitly and all the helpers/tools will work seamlessly. 

But at the moment you start getting "off the rails", you might have a cascade of unexpected complexity that arise :D 

Of course, when needed everything is possible, it's just a tradeoff to arbitrate each time ;-)